### ISSUE TITLE
Document pip installation dependencies (for Fedora 25)

### ISSUE THREAD

**Issue by radomirbosak** [jrnl-org/jrnl#475]
On Fedora 25, I needed to install these dependencies before the installation via pip was successful:
* gcc
* redhat-rpm-config
* python-devel
* python3-devel

These are the error messages that I got
```
unable to execute 'gcc': No such file or directory
```
```
gcc: error: /usr/lib/rpm/redhat/redhat-hardened-cc1: No such file or directory
```
```
build/temp.linux-x86_64-2.7/_openssl.c:12:24: fatal error: pyconfig.h: No such file or directory
```

To install the above dependencies, simply run:
```
$ sudo dnf install gcc redhat-rpm-config python-devel python3-devel
```

Note that only one of `python-devel` and `python3-devel` is needed, depending on whether you install using `pip` or `pip3`.

It would be nice if these dependencies were listed in `CONTRIBUTING.md` or `BUILDING.md` or the documentation.

**Comment by maebert** [comment]
Good catch, I'll add that to the documentation!

**Comment by daviscodesbugs** [comment]
Is there an existing issue for other distribution dependencies? Found one for Ubuntu.

**Comment by maebert** [comment]
This should be fixed in 2.0 since `cryptography` ships with a wheel and doesn't require `gcc`. or `python3-devel`.
